exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/ResourceMarshaler.java

```diff
@@ -31,8 +31,8 @@ static ResourceMarshaler create(io.opentelemetry.sdk.resources.Resource resource
 
   private ResourceMarshaler(AttributeMarshaler[] attributeMarshalers) {
     super(calculateSize(attributeMarshalers));
-    serializedResource = new byte[getSerializedSize()];
-    CodedOutputStream output = CodedOutputStream.newInstance(serializedResource);
+    ByteArrayOutputStream bos = new ByteArrayOutputStream(getSerializedSize());
+    CodedOutputStream output = CodedOutputStream.newInstance(bos);
```

Comment:
Nit: If COS buffers aren't removed, you probably still want to define the buffer size here.